src/Illuminate/Support/Stringable.php

```diff
@@ -1194,6 +1194,30 @@ public function whenIsAscii($callback, $default = null)
         return $this->when($this->isAscii(), $callback, $default);
     }
 
+    /**
+     * Execute the given callback if the string is a valid JSON string.
+     *
+     * @param  callable  $callback
+     * @param  callable|null  $default
+     * @return static
+     */
+    public function whenJson($callback, $default = null)
+    {
+        return $this->when($this->isJson(), $callback, $default);
```

Comment:
To allow for the higher order functions that the `when` method is capable of, I suggest changing this to:

```php
return $this->when($this->isJson(), ...func_get_args());
```

This allows the `func_num_args()` call in `when` to properly pick up `$callback` or `$default` being used.

Reply:
Nevermind. I didn't realize the rest of `Stringable` is doing the same as you suggested...

Reply:
@tomirons Considering the rest of the `Stringable` class, you should discard what I was saying. But to clarify what I meant:

When only one method is provided to the `when` (the condition, in this case, `$this->isJson()`), a `HigherOrderWhenProxy` instance is returned. This allows for method calls such as `$conditionableObject->when(true)->prepend('key.')->unless(false)->append('.key');` or something like that. It also allows for `$object->when()->methodThatReturnsTrue()->doThis();` (also with properties)